File: ResourcesTest_GB/src/main/java/com/example/redballtoy/resourcestest_gb/MyAdapter.java
package com.example.redballtoy.resourcestest_gb;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.ArrayList;

public class MyAdapter extends ArrayAdapter<AndroidVersionModelData> {
    public MyAdapter(Context context, ArrayList<AndroidVersionModelData> androidVersionModelData) {
        super(context, 0, androidVersionModelData);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        //получаем данные из источника данных для определенной позиции
        AndroidVersionModelData androidVersionModelData = getItem(position);
        //если нет созданной вьюшки создаем новую
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_android_version, parent, false);
        }
        //convertview это вьюшка (одного) элемента через которые данные из источника конвертируются
        //в строку в listview, находим элементы в этой вьюшке
        TextView tvAndroidNameVersion = convertView.findViewById(R.id.tv_item_version_name);
        ImageView ivAndroidPic = convertView.findViewById(R.id.ivc_android_icon);
        //заполгяем их данными текущей позиции
        tvAndroidNameVersion.setText(androidVersionModelData.versionName);
        ivAndroidPic.setImageResource(androidVersionModelData.versionPicId);
        //возврат вьюшки элемента списка
        return convertView;
    }

}

File: RecyclerViewWithClickInterface/src/main/java/com/example/redballtoy/recyclerviewwithclickinterface/adapter/MyAdapter.kt
package com.example.redballtoy.recyclerviewwithclickinterface.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.redballtoy.recyclerviewwithclickinterface.R
import com.example.redballtoy.recyclerviewwithclickinterface.model.Model

class MyAdapter(
    private val context: Context,
    private val list: ArrayList<Model>,
    private val cellItemId :Int,
    private val cellClickListener: CellClickListener
) :
    RecyclerView.Adapter<MyAdapter.MyViewHolder>() {

    class MyViewHolder(view: View) :
        RecyclerView.ViewHolder(view) {
        val iconTV = view.findViewById<ImageView>(R.id.iv_logo)
        val titleTV = view.findViewById<TextView>(R.id.tv_title)
        val subtitleTV = view.findViewById<TextView>(R.id.tv_sub_title)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val viewCell = LayoutInflater.from(context)
            .inflate(cellItemId, parent, false)
        return MyViewHolder(viewCell)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val data = list[position]
        holder.iconTV.setImageDrawable(ContextCompat.getDrawable(context, data.icon))
        holder.titleTV.text = data.title
        holder.subtitleTV.text = data.subtitle
        holder.itemView.setOnClickListener {
            cellClickListener.setOnClickListener(data)
        }
    }

    override fun getItemCount(): Int {
        return list.count()
    }


}


